infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/environment/OpenShiftEnvironmentFactory.java

```diff
@@ -122,6 +130,32 @@ protected OpenShiftEnvironment doCreate(
       }
     }
 
+    if (deployments.size() + pods.size() > 1) {
+      List<PodData> podsData =
+          Stream.concat(
+                  pods.values().stream().map(PodData::new),
+                  deployments.values().stream().map(PodData::new))
+              .collect(Collectors.toList());
+
+      Deployment deployment = podMerger.merge(podsData);
+      String deploymentName = deployment.getMetadata().getName();
+
+      // provision merged deployment instead of recipe pods/deployments
+      pods = new HashMap<>();
+      deployments = ImmutableMap.of(deploymentName, deployment);
+
+      // multiple pods/deployments are merged to one deployment
+      // to avoid issues because of overriding labels
+      // provision const label and selector to match all services to merged Deployment
+      deployment
+          .getSpec()
+          .getTemplate()
+          .getMetadata()
+          .getLabels()
+          .put("deployment", deploymentName);
```

Comment:
This part differs from the equal part in K8s implementation. Here literals instead of constants are used. Can we align them?

Reply:
Good catch. Fixed.